horizons/world/player.py

```diff
@@ -69,7 +69,8 @@ def initialize(self, inventory):
 
 	def __init(self, name, color, clientid, difficulty_level, max_tier_notification, settlerlevel=0):
 		assert isinstance(color, Color)
-		assert isinstance(name, basestring) and name
+		assert isinstance(name, basestring)
+		assert name
```

Comment:
I split only `assert isinstance(name, basestring) and name` in two assert.
If you want, I can add.